doc/develop/coding-guidelines.rst

```diff
@@ -0,0 +1,213 @@
+
+For e.g. 1.2 the naming of a base class test is as follows
+
+.. code-block:: python
+
+    class LakeshoreBaseTest:
+        """Base class for testing different models of LakeShore temperature controller."""
+
+        @classmethod
+        def tearDownClass(cls):
+            cls.dev.terminate()  # free up socket.
+
+In class, mark the attributes as private with a first letter as _. They are
+accessible normally from other classes, but by convention, as developers we know that they are
+private to the classes and should not be externally modified.
+If there are public attributes, the attributes might have to be changed in several
+files and folders.
+
+The usage of public attributes is useful when their access is required by a third
+party or another class. In this scenario, public attributes can be used.
+
+For e.g. 1.2 the implementation of public and private attributes
+
+.. code-block:: python
+
+    class Pokemon:
+        """
+        Links special powers to the given pokemon type
+
+        """
+       def __init__(self, name, alias):
+          self.name = name       # public
+          self._alias = alias   # private nomenclature
+
+       def print_who(self):
+          print('name  : ', self.name)
+          print('alias : ', self._alias)
+
+
+Docstrings
+==================
+
+Use three double quotes for the docstring. The docstring consists of function definition, description of input and return arguments. We will follow the reStructuredText style to write the docstring.
+For e.g. 2.1 when type is declared in the docstring
+
+.. code-block:: python
+
+    def get_equality(arg1, arg2):
+        """Summary line.
+
+        Extended description of function.
+
+        :param arg1: (int) Description of arg1.
+        :param arg2: (str) Description of arg2.
+        :raise: ValueError if arg1 is equal to arg2
+        :return: (bool) Description of return value
+
+        """
+
+        if arg1 == arg2:
+            raise ValueError('arg1 must not be equal to arg2')
+
+        return True
+
+
+Either declare the type in the function or in the docstring.
+For e.g. 2.2 when type is not declared in the docstring, it is declared in the function as seen below
+
+.. code-block:: python
+
+    def calculate_mean_and_std(values: List[float]) -> Tuple[float, float]:
+        """
+        Calculates the mean and standard deviation of a list of values
+
+        :param values: list of values.
+        :return: mean and standard deviation
+        """
+
+        mean = sum(values) / len(values)
+        std = () ** 0.5
+        return mean, std
+
+
+In test classes, it is recommended to have one-line docstring for each test method. For short and obvious tests, it is
+not clear if we will use the docstrings.
+
+For e.g. 2.3 the test class should be defined as
+
+.. code-block:: python
+
+    class TestOrsayStatic(unittest.TestCase):
+        """
+        Tests which don't need an Orsay component ready
+        """
+
```

Comment:
@nandishjpatel  in regards to having quotes in the same line, I saw some code in the repository like this. Note sure what is the preferred method.